spec/product_repository_spec.rb

```diff
@@ -0,0 +1,62 @@
+  require 'product_repository'
+
```

Comment:
With the product repository , I like how harry has designed his code, its very readable and clear